Forta-Bots-Challenge-3/src/utils.ts

```diff
@@ -0,0 +1,87 @@
+import { ethers, Contract, BigNumber } from "ethers";
+import { Finding, FindingSeverity, FindingType } from "forta-agent";
+import { L2_FUNCTION_SIGNATURE, L2_TOKEN_ADDRESS_MAKER_DAO } from "./constants";
+
+export const getL1Finding = async (
+  daiContract: Contract,
+  blockNumber: number,
+  escrowAddressArbitrum: string,
+  escrowAddressOptimism: string,
+) => {
+  const arbitrumBalance: BigNumber = await daiContract.balanceOf(
+    escrowAddressArbitrum,
+    {
+      blockTag: blockNumber,
+    },
+  );
+  const optimismBalance: BigNumber = await daiContract.balanceOf(
+    escrowAddressOptimism,
+    {
+      blockTag: blockNumber,
+    },
+  );
+
+  return Finding.fromObject({
+    name: `Combined supply of optimism and Arbitrum MakerDao escrows on layer 1`,
+    description: `Escrow balances: Arbitrum: ${arbitrumBalance} Optimism: ${optimismBalance}`,
+    alertId: "new block check escrows on layer 1",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "Ethereum",
+    metadata: {
+      escrowBalanceOptimism: `${optimismBalance}`,
+      escrowBalanceArbitrum: `${arbitrumBalance}`,
+    },
+  });
+};
+
+export const checkBlock = async (
+  provider: ethers.providers.Provider,
+  blockNumber: number,
+  chainId: number,
+  alert: any,
```

Comment:
Please note that when you call this function, you're passing "alerts" which is an object (of type `AlertsReponse` that contains an array of alerts (`Alert[]`). The way the alerts processed needs to be updated.

Reply:
@Joshua-Medvinsky, this note still holds. 

Also note that you should use only the latest of the alerts fetched (for this you could check the docs for which parameters the `getAlerts` can take)